torch_em/data/segmentation_dataset.py

```diff
@@ -26,15 +26,15 @@ def compute_len(shape, patch_shape):
     def __init__(
         self,
         raw_path: Union[List[Any], str, os.PathLike],
-        raw_key: str,
+        raw_key: Optional[str],
         label_path: Union[List[Any], str, os.PathLike],
-        label_key: str,
+        label_key: Optional[str],
         patch_shape: Tuple[int, ...],
         raw_transform=None,
         label_transform=None,
         label_transform2=None,
         transform=None,
-        roi: Optional[dict] = None,
+        roi: Optional[Union[Dict[str, Any], List[Any]]] = None,
```

Comment:
I don't think this one is true.  This actually needs to either be a slice or a tuple of slices.